### PR TITLE
[fix] draw full-width chars problem

### PR DESCRIPTION
**Pull request by SeeLog**
I found the following problem.
スクリーンショット 2022-05-30 23 23 52

Half-width characters (such as "123") are painted correctly, but full-width ones(such as "１２３", "あいう") are painted incorrectly(not centering).

I have solved this problem with this diff.
Like this:
スクリーンショット 2022-05-30 23 23 17

If you need it, please refer to it.

### CODE REVIEW

File: lib/src/roulette_paint.dart [hunk 1]
```diff
@@ -127,15 +127,20 @@ class _RoulettePainter extends CustomPainter {
...
       final textStyle = unit.textStyle ?? style.textStyle;
-      final pb = ui.ParagraphBuilder(ui.ParagraphStyle())
+      final pb = ui.ParagraphBuilder(ui.ParagraphStyle(
+        textAlign: TextAlign.center,
+      ))
         ..pushStyle(textStyle.asUiTextStyle())
         ..addText(text);
 
+      // Calculate chord of circle
+      final chord = 2 * (radius * style.textLayoutBias) * sin(sweep / 2);
```
Comment (SeeLog) [inline lib/src/roulette_paint.dart:137]: `radius * style.textLayoutBias` is the y-position of the text that we want to paint.
We can calculate chord length with $2r \sin\theta/2$.(r: radius, $\theta$: radian)

File: lib/src/roulette_paint.dart [hunk 2]
```diff
@@ -127,15 +127,20 @@ class _RoulettePainter extends CustomPainter {
...
+
       final p = pb.build();
-      p.layout(const ui.ParagraphConstraints(width: double.infinity));
+      p.layout(ui.ParagraphConstraints(width: chord));
```
Comment (SeeLog) [inline lib/src/roulette_paint.dart:140]: Limit paragraph area with chord length.

File: lib/src/roulette_paint.dart [hunk 3]
```diff
@@ -127,15 +127,20 @@ class _RoulettePainter extends CustomPainter {
...
 
       canvas.drawParagraph(
-          p, Offset(-p.minIntrinsicWidth / 2, -radius * style.textLayoutBias));
+          p, Offset(-chord / 2, -radius * style.textLayoutBias));
```
Comment (SeeLog) [inline lib/src/roulette_paint.dart:143]: This means the center of the limited area.

**Comment by do9core** [comment]
Thanks for the fix! I'll check it.

**Review by do9core — APPROVED** [review]

**Comment by do9core** [comment]
Seems these changes affected the presentation test, could you update the golden files?
The golden files were generated on MacOS.
If you have any trouble on it, I can handle it.

**Comment by SeeLog** [comment]
Thank you for your review!!

I have added a fixed golden file.
Now, I confirm to pass all tests.

**Comment by do9core** [comment]
I'll published a new release soon, thanks again for your contribution. 😊

**Pull request merged**